includes/utils/class-amp-image-list.php

```diff
@@ -0,0 +1,138 @@
+<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
+
+// phpcs:disable Generic.Files.OneObjectStructurePerFile.MultipleFound
+/**
+ * Image collection classes.
+ *
+ * @package AMP
+ */
+
+/**
+ * Interface Has_Caption
+ *
+ * @since 1.5.0
+ */
+interface Has_Caption {
+	/**
+	 * Gets the caption.
+	 *
+	 * @return string The caption text.
+	 */
+	public function get_caption();
+}
+
+/**
+ * Class AMP_Image
+ *
+ * @since 1.5.0
+ */
+class AMP_Image {
+
+	/**
+	 * The image node.
+	 *
+	 * @var DOMElement
+	 */
+	protected $image_node;
+
+	/**
+	 * Constructs the class.
+	 *
+	 * @param DOMElement $image_node The image node.
+	 */
+	public function __construct( DOMElement $image_node ) {
+		$this->image_node = $image_node;
+	}
+
+	/**
+	 * Gets the image.
+	 *
+	 * @return DOMElement
+	 */
+	public function get_image_node() {
+		return $this->image_node;
+	}
+}
+
+/**
+ * Class AMP_Captioned_Image
+ *
+ * @since 1.5.0
+ */
+final class AMP_Captioned_Image extends AMP_Image implements Has_Caption {
```

Comment:
Should be in a separate file.

Reply:
Sure, this is now [in its own file](https://github.com/ampproject/amp-wp/pull/3659/files#diff-672fa2445cf206c6ba3b93fca58976c0R1).